pm/recipient.go

```diff
@@ -363,16 +366,24 @@ func (r *recipient) updateSenderNonce(rand *big.Int, ticket *Ticket) error {
 	defer r.senderNoncesLock.Unlock()
 
 	randStr := rand.String()
-	sn, ok := r.senderNonces[randStr]
-	if ok && ticket.SenderNonce <= sn.nonce {
-		return errors.Errorf("invalid ticket senderNonce sender=%v nonce=%v highest=%v", ticket.Sender.Hex(), ticket.SenderNonce, sn.nonce)
+	senderNonce, randKeySeen := r.senderNonces[randStr]
+	if randKeySeen {
+		_, isSeen := senderNonce.nonceSeen[ticket.SenderNonce]
+		if isSeen {
+			return errors.Errorf("invalid ticket senderNonce: already seen sender=%v nonce=%v", ticket.Sender.Hex(), ticket.SenderNonce)
+		}
+	} else {
+		r.senderNonces[randStr] = &struct {
+			nonceSeen       map[uint32]byte
+			expirationBlock *big.Int
+		}{make(map[uint32]byte), ticket.ParamsExpirationBlock}
 	}
-
-	r.senderNonces[randStr] = &struct {
-		nonce           uint32
-		expirationBlock *big.Int
-	}{ticket.SenderNonce, ticket.ParamsExpirationBlock}
-
+	// check nonce map size
+	if len(r.senderNonces[randStr].nonceSeen) > maxSenderNonces-1 {
```

Comment:
```suggestion
	if len(r.senderNonces[randStr].nonceSeen) >= maxSenderNonces {
```

Nit: IMO feels a bit clearer (no arithmetic needed) and conveys "if we are at the max size or for some reason exceeded it, return an error".